protowhat/selectors.py

```diff
@@ -8,24 +8,23 @@
 
 
 class Selector(NodeVisitor):
-    def __init__(self, target_cls, target_cls_name=None, strict=True, priority=None):
+    def __init__(self, target_cls, target_cls_name=None, strict=True, priority=None, include_head=False):
         self.target_cls = target_cls
         self.target_cls_name = target_cls_name
         self.strict = strict
         self.priority = priority if priority else self._get_node_priority(target_cls)
         self.out = []
+        self.include_head = include_head
 
     def visit(self, node, head=False):
         """
         Find child nodes at the first level
-        and keep searching if their children have a lower priority
+        and keep searching if their children have a lower priority.
+        If self.include_head is True then the starting node will also be considered.
         """
-        if head:
-            return super().visit(node)
-
-        if self.is_match(node):
+        if self.include_head and self.is_match(node):
```

Comment:
```suggestion
        if (not head or self.include_head) and self.is_match(node):
```

Reply:
Yes you're right! Saw the test failing and came to the same solution.